Made alterations to deal with Codacy code style issues

rowan08 opened the pull request:
Passed several variables by reference in functions, where possible
Changed variable initialization to use an initialization list
Initialized variable `_isNull` to true in Field class
Changed negation operations to explicit form in if-statements
Tested with make
Issue: https://github.com/davidbrownza/CORMAP/issues/41

rowan08 committed:
Made alterations to deal with Codacy code style issues
Passed several variables by reference in functions, where possible
Changed variable initialization to use an initialization list
Initalized variable _isNull to true
Changed negation operations to explicit form
Tested with make
Issue: https://github.com/davidbrownza/CORMAP/issues/41

davidbrownza commented on src/fields/Field.cpp:
```diff
@@ -1,13 +1,9 @@
-}
+Field::Field(string &fieldName, FieldType fieldType, bool isPrimaryKey, bool isUnique, bool isNullable, bool isAutoFilled) :
+    _fieldName(fieldName), _fieldType(fieldType), _isPrimaryKey(isPrimaryKey), _isUnique(isUnique), _isNullable(isNullable), _isAutoFilled(isAutoFilled) {
+        _isNull = true;
```
Can `isNull` not also be initialized in the list?

rowan08 replied:
It can, but g++ gives a warning:
```
warning: ‘Field::_isAutoFilled’ will be initialized after [-Wreorder]
         bool _isAutoFilled;
              ^
../src/fields/Field.h:22:14: warning:   ‘bool Field::_isNull’ [-Wreorder]
         bool _isNull;
```
Apart from this it does run fine.

rowan08 replied:
Ok, never mind. This had to do with the order these appeared in `Field.h`, so it can also be initialized in the list.
Done

davidbrownza commented on src/fields/CharField.cpp:
```diff
@@ -1,11 +1,8 @@
-    _fieldValue = defaultValue;
-    _maxLength = maxLength;
-}
+CharField::CharField(string &fieldName, int maxLength, string defaultValue, bool isPrimaryKey, bool isUnique, bool isNullable):
```
Why pass in `fieldName` by reference, but not `defaultValue`

davidbrownza commented on src/fields/TextField.cpp:
```diff
@@ -1,9 +1,7 @@
-    Field(fieldName, TEXT, isPrimaryKey, isUnique, isNullable) {
-    _fieldValue = defaultValue;
-}
+TextField::TextField(string &fieldName, string defaultValue, bool isPrimaryKey, bool isUnique, bool isNullable):
```
Again, why not pass in `defaultValue` by reference?

rowan08 replied:
The problem here is that in `TextField.h`/`CharField.h`, this has a value `defaultValue=""`.
So It throws an error if you try to pass by reference. I still need to see if I can find a way around this...

davidbrownza commented on src/model/Model.cpp:
```diff
@@ -1,12 +1,11 @@
 
 Model::~Model() {
-    for (vector<Field*>::const_iterator it = fields.begin(); it != fields.end(); it++) {
+    for (vector<Field*>::const_iterator it = fields.begin(); it != fields.end(); ++it) {
```
This likely to be a breaking change as it changes the logic. Putting the ++ before `it` means that it increments before going into the loop, thereby skipping the first `Field` in the vector. It would also likely be a very difficult bug to find in the future.

rowan08 replied:
From what I can tell, this doesn't apply to for-loops declaration, unless you did something like:
`for (int i=0, a=0; i < 10; a = i++) {` vs `for (int j=0, a=0; j < 10; a = ++j) {`
In which case `a` would be affected, but `i`/`j` are equal in both (I may be wrong though).

Otherwise, do we just leave this as it was and ignore the code styling issue (Prefer prefix ++/-- operators for non-primitive types.) or is there another way around this?

davidbrownza replied:
Have you tested it?

rowan08 replied:
Yes, I copy/pasted the for-loops above from my test script. Only the value of `a` changed between the two.

davidbrownza replied:
Okay cool, keep it this way then.

davidbrownza commented on src/model/Model.cpp:
```diff
@@ -179,7 +178,7 @@ void Model::generateSQLParts(string &sqlHead, string &sqlBatch, string &sqlUpdat
 }
 
-string Model::generateSQLTail(string sqlBatch, string sqlUpdate, int batchsize, Mode mode) {
+string Model::generateSQLTail(string sqlBatch, string &sqlUpdate, int batchsize, Mode mode) {
```
Why only `sqlUpdate` as reference?

rowan08 replied:
Hmm, not sure how I missed that. 
Have updated this so that `sqlBatch `and `sqlUpdate `are both passed by reference

rowan08 committed:
Undid pass-by-reference in Fields classes.
Added additional pass-by-reference in Model class.
Fixed _isNull to be initialized in initialization list.
Tested with make.
Issue: https://github.com/davidbrownza/CORMAP/issues/41
